tests/pytest_tests/conftest.py

```diff
@@ -20,6 +21,50 @@
 from wandb.sdk.lib.gitlib import GitRepo  # noqa: E402
 from wandb.sdk.lib.paths import StrPath  # noqa: E402
 
+# --------------------------------
+# Global pytest configuration
+# --------------------------------
+
+
+@pytest.fixture(autouse=True)
+def toggle_wandb_core(
+    monkeypatch: pytest.MonkeyPatch,
+    request: pytest.FixtureRequest,
+) -> None:
+    """Sets WANDB__REQUIRE_CORE in each test.
+
+    wandb-core is a Go rewrite of some Python logic, and all tests should work
+    the same both with and without it.
+    """
+    monkeypatch.setenv("WANDB__REQUIRE_CORE", str(request.param))
```

Comment:
can you use context manager with yield? t

Reply:
i think this is better, since it's one line and `monkeypatch` is specifically supported by `pytest`

Reply:
ok, tho i remember we had some issues with this patcher